### PR TITLE
use rapids-build-backend

### PR DESCRIPTION
**Pull request by jameslamb**
Contributes to https://github.com/rapidsai/build-planning/issues/31
Contributes to https://github.com/rapidsai/dependency-file-generator/issues/89

Proposes introducing `rapids-build-backend` as this project's build backend, to reduce the complexity of various CI/build scripts.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -111,6 +111,7 @@ if (( ${NUMARGS} == 0 )) || hasArg cuxfilter; then
     echo "8"
     if [[ ${INSTALL_TARGET} != "" ]]; then
         python setup.py build_ext --inplace
+        RAPIDS_DISABLE_CUDA=true \
         python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (bdice) [inline build.sh:115]: I think we usually indent the line following an explicit break like this.
```suggestion
        RAPIDS_DISABLE_CUDA=true \
            python setup.py install --single-version-externally-managed --record=record.txt
```

**Review by bdice — APPROVED** [review]
LGTM, just one minor comment.

File: build.sh (hunk 1 above)
Comment (jameslamb) [reply build.sh:115]: ok sure

**Review by AjayThorve — APPROVED** [review]

**Pull request merged**